test/merge-patient-lab-iips-mocha.js

```diff
@@ -0,0 +1,138 @@
+                    return component.rdfpf.createVni('').getInputState('labwork', '', '');
+                }).then(_.property('data')).should.become({id: '001',  glucose: '75',  date: '2012-02-01'});
+            });
+        });
+        describe('#getOutputState', function() {
+            it("should initially return nothing", function() {
+                return Promise.resolve(component).then(function(component){
+                    return component.rdfpf.createVni('').getOutputState();
+                }).should.become(undefined);
+            });
+            it("should have a output state after input", function() {
+                return Promise.resolve(component).then(function(component){
+                    var patient = noflo.internalSocket.createSocket();
+                    var labwork = noflo.internalSocket.createSocket();
+                    component.inPorts.patient.attach(patient);
+                    patient.send({id: '001',  name: 'Alice', dob: '1979-01-23' });
+                    patient.disconnect();
+                    component.inPorts.labwork.attach(labwork);
+                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
+                    labwork.disconnect();
+                    component.inPorts.patient.detach(labwork);
+                    component.inPorts.patient.detach(patient);
+                    return component.rdfpf.createVni('').getOutputState();
+                }).then(_.keys).then(_.sortBy).should.become(_.sortBy(['data','lm', 'previousLms', 'selfPort']));
+            });
+            it("should have an output state data after input", function() {
+                return Promise.resolve(component).then(function(component){
+                    var patient = noflo.internalSocket.createSocket();
+                    var labwork = noflo.internalSocket.createSocket();
+                    component.inPorts.patient.attach(patient);
+                    patient.send({id: '001',  name: 'Alice', dob: '1979-01-23' });
+                    patient.disconnect();
+                    component.inPorts.labwork.attach(labwork);
+                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
+                    labwork.disconnect();
+                    component.inPorts.patient.detach(labwork);
+                    component.inPorts.patient.detach(patient);
+                    return component.rdfpf.createVni('').getOutputState();
+                }).then(_.property('data')).should.become({id: '001',  name: 'Alice', dob: '1979-01-23', glucose: '75',  date: '2012-02-01'});
+            });
+        });
+    });
+});
+
+function createComponent(getComponent) {
+    var component = getComponent();
+    _.forEach(component.inPorts, function(port, name) {
+        port.nodeInstance = component;
+        port.name = name;
+    });
+    _.forEach(component.outPorts, function(port, name) {
+        port.nodeInstance = component;
```

Comment:
Wouldn't the test be more realistic if a NoFlo server were run, instead of manually imitating what NoFlo does?  Or would that make testing too difficult?